modules/collections/client/controllers/collections.controllers.js

```diff
@@ -0,0 +1,344 @@
+          animation: true,
+          templateUrl: 'modules/utils/client/views/partials/modal-error.html',
+          controller: function($scope, $state, $uibModalInstance) {
+            var self = this;
+            self.title = title || 'An error has occurred';
+            self.msg = msg;
+            self.ok = function() {
+              $uibModalInstance.close($scope.org);
+            };
+            self.cancel = function() {
+              $uibModalInstance.dismiss('cancel');
+            };
+          },
+          controllerAs: 'self',
+          scope: $scope,
+          size: 'md',
+          windowClass: 'modal-alert',
+          backdropClass: 'modal-alert-backdrop'
+        });
+        // do not care how this modal is closed, just go to the desired location...
+        modalDocView.result.then(function (/* res */) {transitionCallback(); }, function (/* err */) { transitionCallback(); });
+      };
+
+      $scope.confirmMove = function(title, msg, moveFunc) {
+        var modalDocView = $uibModal.open({
+          animation: true,
+          templateUrl: 'modules/utils/client/views/partials/modal-confirm-generic.html',
+          controller: function($scope, $state, $uibModalInstance) {
+            var self = this;
+            self.title = title || 'Move document?';
+            self.question = msg || 'Are you sure?';
+            self.actionOK = 'Move';
+            self.actionCancel = 'Cancel';
+            self.ok = function() {
+              $uibModalInstance.close($scope.project);
+            };
+            self.cancel = function() {
+              $uibModalInstance.dismiss('cancel');
+            };
+          },
+          controllerAs: 'self',
+          scope: $scope,
+          size: 'md',
+          windowClass: 'modal-alert',
+          backdropClass: 'modal-alert-backdrop'
+        });
+        modalDocView.result.then(function(/* res */) {
+          moveFunc();
+        });
+      };
+
+      var goToList = function() {
+        $state.transitionTo('p.collection.list', { projectid: project.code }, {
+          reload: true, inherit: false, notify: true
+        });
+      };
+
+      var goToDetail = function() {
+        $state.transitionTo('p.collection.edit', { projectid: project.code, collectionId: collection._id }, {
+          reload: true, inherit: false, notify: true
+        });
+      };
+
+      var reloadEdit = function() {
+      // want to reload this screen, do not catch unsaved changes (we are probably in the middle of saving).
+        $scope.allowTransition = true;
+        $state.reload();
+      };
+
+      $scope.otherDocsReordered = function() {
+        reloadEdit();
+      };
+
+      $scope.goToDocument = function(doc) {
+      // Open document in doc manager.
+        $location.url('/p/' + $scope.project.code + '/docs?folder=' + doc.directoryID);
+      };
+
+      $scope.updateDocuments = function(updatedDocuments, sourceDocuments, docType) {
```

Comment:
Merged _updateMainDocument_ and _updateOtherDocuments_ into one function that flexes based on the type of document being processed (added as a parameter in the function call).